stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -481,6 +481,10 @@ end
 # dismabiguation methods: * of Diagonal and Adj/Trans AbsVec
 *(A::Diagonal, B::Adjoint{<:Any,<:AbstractVector}) = A * copy(B)
 *(A::Diagonal, B::Transpose{<:Any,<:AbstractVector}) = A * copy(B)
-*(A::Adjoint{<:Any,<:AbstractVector}, B::Diagonal) = copy(A) * B
-*(A::Transpose{<:Any,<:AbstractVector}, B::Diagonal) = copy(A) * B
+*(x::Adjoint{<:Any,<:AbstractVector}, D::Diagonal) = Adjoint(map((t,s) -> t'*s, D.diag, parent(x)))
```

Comment:
Perhaps `(D' * x')'` would be simpler / clearer, if methods appropriate for `D' * x'` exist?